adafruit_display_shapes/sparkline.py

```diff
@@ -111,8 +115,14 @@ def add_value(self, value):
     # pylint: disable=no-else-return
     @staticmethod
     def _xintercept(
-        x_1, y_1, x_2, y_2, horizontal_y
-    ):  # finds intercept of the line and a horizontal line at horizontalY
+        x_1: int,
+        y_1: float,
+        x_2: int,
+        y_2: float,
```

Comment:
I think x1, y1 and x2, y2 here are a pair of points in a 2D plane. I think they should match types for all of them. But I'm not certain if `int` or `float` is the right one. I'll look further into it to see if it can be worked out from usages of this function.

Reply:
Thank you for the feedback!

In this case, the function is used like this:

```python
xint_bottom = self._xintercept(
      x_1, last_value, x_2, value, self.y_bottom
)
```

And `value` and `last_value` come from `self._spark_list`, which in one of the examples is populated with float numbers, like this:

```python
sparkline1.add_value(random.uniform(0, 10))
```

I would say, the options could be:

1. Either use float, since [PEP-0484](https://peps.python.org/pep-0484/#the-numeric-tower) suggests that int is acceptable when an argument is annotated as float.
2. Or probably use `Union[int, float]` to document that both types are possible.

Please let me know if one of these would work or if you have a different proposal.

Thank you!

Reply:
Thanks for the additional info and reference link. Lets go with `float` for all of them.

Reply:
Thanks!